pkg/k8s/client/cell.go

```diff
@@ -503,6 +514,37 @@ func NewStandaloneClientset(cfg Config) (Clientset, error) {
 	return clientset, err
 }
 
+type ClientBuilderFunc func(string) Clientset
+
+// NewClientBuilder returns a function that creates a new Clientset with the given
+// name as the user agent, or logs a fatal error if the Clientset cannot be
+// created.
+func NewClientBuilder(lc cell.Lifecycle, log logrus.FieldLogger, cfg Config) ClientBuilderFunc {
+	return func(name string) Clientset {
+		// Note: We are only making a shallow copy of the Config. This will need
+		// to be replaced with a DeepCopy method if any pointer types are
+		// introduced to Config.
+		newCfg := cfg
+		if name != "" {
+			newCfg.UserAgent = name
+		}
+		c, err := newClientset(lc, log, newCfg)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
Just a suggestion that I'd rather return the error here and let the consumer decide what to do, even if, in this specific case, it won't make much of a difference (adapting `registerController` to propagate the error results to a failure in the hive construction with a `panic`). But it might be better for future uses of the ClientBuilder.

Reply:
Good suggestion! done :+1: